proposals/namespace_editing/README.md

```diff
@@ -0,0 +1,480 @@
+
+For instance, if we take the same example in the above section, it's possible to create a UsdEditTarget that targets the model.usda layer using the node that represents the reference to model.usda's "Model" prim in the prim index for /A. This edit target would have a map function that maps /A to /Model like the node does for the reference. Now, if we were to use the UsdNamespaceEditor API to rename /A/Child to /A/Ball on the root.usda stage but using the edit target we just created, this namespace edit would be mapped across the edit target. I.e. the paths processed by the rename operation would be mapped so that the rename will be from /Model/Child to /Model/Ball and the edits would be performed in model.usda. Or to put this yet another way, editing across this edit target would be equivalent to opening model.usda as its own UsdStage and performing the same namespace edit using the paths mapped from /A to /Model.
+
+A few things to note here:
+1. Editing across the reference using an edit target can cause additional side effect namespace changes. In the example, by performing the rename of /A/Child to /A/Ball across the model reference, thereby renaming /Model/Child to /Model/Ball in model.usda, **will cause /B/Child to be renamed to /B/Ball in the root stage**as well. This is a different outcome from when we performed the edit in the root layer stack.
+2. Namespace edits across arcs may require propagating “fixup edits” to stronger layer stacks (which is explained in the next section) in order to maintain composed opinions on the edited object.
+3. It’s also worth noting that as the UsdEditTarget currently only holds a layer and mapping function, we will either need to be able to determine a targeted layer stack from this information or add layer stack identification back into UsdEditTarget to be used during namespace editing.
+
+Fixup of Namespace Targets
+--------------------------
+
+As mentioned earlier, one major consideration when handling namespace editing in USD is that we have ways in which scene description can refer to or target namespace paths of other objects. Any scene description that targets another path may become invalid if a namespace edit were to change or remove the path of the targeted object. Thus, we will provide the ability to automatically author the necessary edits to fix up paths to namespace objects that have been moved or deleted through the UsdNamespaceEditor API.
+
+The types of paths in scene description that we will need to be able to fix up are:
+* Relationship targets
+* Attribute connections
+* Path expressions for pattern based collections
+* Composition arcs (references, payloads, inherits, specializes)
+* Overs in higher composition strength layer stacks
+* Layer metadata for default prim (which can be consumed by references and payloads)
+* Possibly internal path reliant structures (like a stage's current edit target which may have a path map function)
```

Comment:
We may want to note that UsdCollectionAPI::MembershipQuery objects retained by clients may need to be discarded in response to namespace edits, as well.

Reply:
That's another good example; I'll add it to that bullet.